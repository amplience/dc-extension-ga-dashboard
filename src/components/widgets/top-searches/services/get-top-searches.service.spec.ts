import getTopSearches, {
  getTopSearchesForCsv,
} from './get-top-searches.service';
import type { SearchIndex } from 'dc-management-sdk-js';
import topSearchesFixture from '../__fixtures__/top-search-single-item-list';

describe('getTopSearches', () => {
  it('should list top hits for a given search index and params', async () => {
    const mockedGetItems = jest
      .fn()
      .mockImplementation(() => topSearchesFixture);
    const mockedTopSearchesGet = jest.fn().mockImplementation(() => {
      return {
        getItems: mockedGetItems,
      };
    });
    const mockedSearchIndex = {
      related: {
        'top-searches': {
          get: mockedTopSearchesGet,
        },
      },
    };
    const params = {
      limit: 5,
      clickAnalytics: true,
      includeReplicas: true,
      startDate: '2020-01-01',
      endDate: '2020-01-01',
    };

    const result = await getTopSearches(
      (mockedSearchIndex as unknown) as SearchIndex,
      params
    );

    expect(mockedTopSearchesGet).toHaveBeenCalledWith(params);
    expect(mockedGetItems).toHaveBeenCalled();
    expect(result).toMatchSnapshot();
  });
});

describe('getTopSearchesForCsv', () => {
  it('should only list csv specific properties for top searches for a given search index and params', async () => {
    const mockedGetItems = jest
      .fn()
      .mockImplementation(() => topSearchesFixture);
    const mockedTopSearchesGet = jest.fn().mockImplementation(() => {
      return {
        getItems: mockedGetItems,
      };
    });
    const mockedSearchIndex = {
      related: {
        'top-searches': {
          get: mockedTopSearchesGet,
        },
      },
    };
    const params = {
      clickAnalytics: true,
      includeReplicas: true,
      startDate: '2020-01-01',
      endDate: '2020-01-01',
    };

    const result = await getTopSearchesForCsv(
      (mockedSearchIndex as unknown) as SearchIndex,
      params
    );

    expect(result).toMatchSnapshot();
  });
});
