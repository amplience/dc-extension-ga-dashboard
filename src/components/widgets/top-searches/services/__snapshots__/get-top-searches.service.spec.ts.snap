// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTopSearches should list top hits for a given search index and params 1`] = `
Array [
  Object {
    "averageClickPosition": 2.5,
    "clickCount": 2,
    "clickPositions": Array [
      Object {
        "clickCount": 3,
        "position": Array [
          1,
          1,
        ],
      },
      Object {
        "clickCount": 0,
        "position": Array [
          2,
          2,
        ],
      },
      Object {
        "clickCount": 0,
        "position": Array [
          3,
          3,
        ],
      },
      Object {
        "clickCount": 0,
        "position": Array [
          4,
          4,
        ],
      },
      Object {
        "clickCount": 0,
        "position": Array [
          5,
          5,
        ],
      },
      Object {
        "clickCount": 4,
        "position": Array [
          6,
          6,
        ],
      },
      Object {
        "clickCount": 0,
        "position": Array [
          7,
          7,
        ],
      },
      Object {
        "clickCount": 0,
        "position": Array [
          8,
          8,
        ],
      },
      Object {
        "clickCount": 0,
        "position": Array [
          9,
          9,
        ],
      },
      Object {
        "clickCount": 0,
        "position": Array [
          10,
          10,
        ],
      },
      Object {
        "clickCount": 0,
        "position": Array [
          11,
          20,
        ],
      },
      Object {
        "clickCount": 0,
        "position": Array [
          21,
          -1,
        ],
      },
    ],
    "clickThroughRate": 0.25,
    "conversionCount": 2,
    "conversionRate": 0.25,
    "count": 11,
    "nbHits": 9,
    "search": "abc",
    "trackedSearchCount": 8,
  },
]
`;

exports[`getTopSearchesForCsv should only list csv specific properties for top searches for a given search index and params 1`] = `
Array [
  Object {
    "averageClickPosition": 2.5,
    "clickThroughRate": 0.25,
    "conversionRate": 0.25,
    "count": 8,
    "query": "abc",
  },
]
`;
